metabase/metabase
# Fix crash when navigating between bookmarked dashboards

Fixes #47170

The PR updates the dashboard fetching logic, so it no longer calls `setErrorPage` for canceled request exceptions.

Would appreciate ideas on test coverage for this: I believe a Cypress test would be too flaky for bugs like that, and I failed to tune request timing/sequencing in a Jest test for this :/

### To verify

1. Create and bookmark 2 dashboards
2. Open app sidebar
3. Throttle your network (or use stats instance) or be quick with the next steps
4. Navigate to 1st dashboard
5. Open app sidebar
6. Navigate to 2nd dashboard
7. Ensure there're no visible errors

### Demo

**Before**

https://github.com/user-attachments/assets/0f1d7ff3-3f04-4683-951a-534e02de5895

**After**

https://github.com/user-attachments/assets/abec24e1-5b90-4086-a240-e7e92a14a404
